### PR TITLE
Milestone 6

### PR DESCRIPTION
**Pull request by Tripple-A**
WE have changed the friends method for the user such that every user friendship relationship has two rows... we have also included tests to ensure the new methods work.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  has_many :posts
+  has_many :likes, class_name: 'Like', foreign_key: 'liker_id'
+  has_many :comments, class_name: 'Comment', foreign_key: 'commenter_id'
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friendships.map { |friendship| friendship.friend if friendship.confirmed }
```
Comment (adriaanbd) [inline app/models/user.rb]: This doesn't seem to be working properly (unless I'm missing something here):

```
irb(main):016:0> u2.confirm_friend(u1)
   (7.8ms)  COMMIT
=> #<Friendship id: 5, user_id: 2, friend_id: 1, confirmed: true, created_at: "2019-12-20 04:58:01", updated_at: "2019-12-20 04:58:01">
```

I use the method and returns nil. Shouldn't it return the friends with a `confirmed: true` status?

```
irb(main):026:0> u2.friends
=> [nil]
irb(main):027:0> u1.friends
=> [nil, nil, nil]
```

[`REQUIRED`]

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  def delete_friend(user)
+    request1 = Friendship.where(user_id: user.id, friend_id: id).first
+    request2 = Friendship.where(user_id: id, friend_id: user.id).first
+    a = request1.destroy if request1
+    b = request2.destroy if request2
+    a || b
+  end
+
+  def friend?(user)
```
Comment (adriaanbd) [inline app/models/user.rb]: This isn't working properly either

There is a friendship, user 1 and friend 2
```
irb(main):036:0> Friendship.find_by(confirmed: true)
 => #<Friendship id: 1, user_id: 1, friend_id: 2, confirmed: true, created_at: "2019-12-20 04:51:46", updated_at: "2019-12-20 04:58:01">
```

Let's use the method to see:

```
irb(main):033:0> u1.friend?(u2)
=> false
irb(main):035:0> u2.friend?(u1)
=> false
```

But I thought they were friends, right?

[`REQUIRED`]

Comment (Tripple-A) [reply app/models/user.rb]: friendship working perfectly in my console milestone6
I have attached a screenshot of friendships working perfectly in my console without any changes made. You have nil in an empty array,it is possible you already deleted the friendships at some point,pls check as my screenshot shows no errors whatsoever in the relationships. Also,concerning the issue raised about friendhips being able to be created,as seen in the screenshot below,in the app,it is impossible to create a friendship twice as it is prohibited from the create method in the controller. To see this work in the console as well,I believe you have to call that controller method in the console. Please test this in the app and verify that this is impossible as it truly is,I believe users who want to create friendships would use the app. Thank you.
friendshipscontroller#create

**Comment by Tripple-A** [comment]
## WE have had a chat with the previous TSE and the issue raised has been hashed out ,PLease confirm the inability of requests to be created from both ends from the app as we have found that prohibiting the database would interfer with majority of our methods,hence our decision to implement it from the controller. Thank you.

**Pull request closed without merging**